### DISCUSSION TITLE
About useIsomorphicLayoutEffect

### DISCUSSION THREAD

**Discussion by tianzhich** [reduxjs/react-redux#1680] [General]
Hi, I had a question about `useIsomorphicLayoutEffect` in `useSelector`. Is there any difference to use `useLayoutEffect` instead of `useEffect`? 

I found the [comment](https://github.com/reduxjs/react-redux/blob/master/src/utils/useIsomorphicLayoutEffect.js) which said:

> // React currently throws a warning when using useLayoutEffect on the server.
// To get around it, we can conditionally useEffect on the server (no-op) and
// useLayoutEffect in the browser. We need useLayoutEffect to ensure the store
// subscription callback always has the selector from the latest render commit
// available, otherwise a store update may happen between render and the effect,
// which may cause missed updates; we also must ensure the store subscription
// is created synchronously, otherwise a store update may occur before the
// subscription is created and an inconsistent state may be observed

I know the `useLayoutEffect` extends the render phase of a component. But if used just in a custom hook like `useSelector`. Must we need `useLayoutEffect`?

**Comment by tianzhich** [comment]
In other words, in custom hook, is there any difference between `useLayoutEffect` and `useEffect`?

**Comment by markerikson** [comment]
I'm not sure what you're asking here.  Are you asking specifically about how and why React-Redux has to write the logic this way, or whether you should call `useLayoutEffect` in your own code?

**Reply by tianzhich** [reply]
Thanks for your reply, and yes, it's about how and why React-Redux has to write the logic this way
